erpnext/stock/doctype/warehouse/warehouse_tree.js

```diff
@@ -23,5 +23,19 @@ frappe.treeview_settings['Warehouse'] = {
 			+ format_currency(Math.abs(node.data.balance), node.data.company_currency)
 			+ '</span>').insertBefore(node.$ul);
 		}
-	}
+	},
+	toolbar: [
+		{
+		  label:__("Add Child"),
+		  condition: function(node) {
+			return node.expandable;
+		  },
+		  click: function(node) {
+			frappe.new_doc("Warehouse",{
+				company: cur_tree.args.company,
+			})
+		  }
+		}
+	  ],
+	  extend_toolbar: true
```

Comment:
```suggestion
	},
```
@nmimsnikhil can we make it more generic.  instead of overriding toolbar modify it in treeview.js. we have tree more doctype like account, cost center and warehouse and more. just pass company args to new.doc in treeview.js will get updated every tree add child trigger.

Reply:
@vishdha Modifying directly in treeview.js is not an ideal situation since treeview is a part of framework and not erpnext.

Reply:
@hrwX Yeah Right.